pkg/cli/mirror/catalog_images.go

```diff
@@ -223,8 +268,10 @@ func (o *MirrorOptions) processCatalogRefs(ctx context.Context, catalogsByImage
 				containertools.ConfigsLocationLabel: "/configs",
 			}
 			cfg.Config.Labels = labels
-			cfg.Config.Cmd = []string{"serve", "/configs"}
-			cfg.Config.Entrypoint = []string{"/bin/opm"}
+			// Although it was prefered to keep the entrypoint and command as it was
+			// we couldnt guarantie that the cache-dir was /tmp/cache, and therefore
+			// we had to specify the command for the newly built catalog
+			cfg.Config.Cmd = []string{"serve", "/configs", "--cache-dir=/tmp/cache"}
```

Comment:
See my other comment about cache dir discovery. I wonder if this needs to be more selective about the arguments you update here. This just completely overwrites the command with hard coded values... what if future versions of opm need different arguments to start?

Reply:
If we're overriding the --cache-dir argument always, then maybe the discovery part does not matter? Something to ponder.

Reply:
Sorry I don't think I was clear enough. I think that this code should find `--cache-dir` and **_only modify its contents_** and leave the rest of the arguments alone. As I mentioned in https://github.com/openshift/oc-mirror/pull/651#discussion_r1284725550 there's two ways the argument could be potentially specified:

"--cache-dir=/tmp/cache"
"--cache-dir", "/tmp/cache"

The reason for this is to protect the "serve" arguments in case they change in the future (e.g. adding a new positional argument or flag argument). Hard coding `[]string{"serve", "/configs", "--cache-dir=/tmp/cache"}` would break any future changes that might be made to the serve command.

My second comment regarding "discovery" was about trying to see what the current value for `--cache-dir` is and using that path when we change `cfg.Config.Cmd`. In my mind that's optional because if we just change to `/tmp/cache` always then the discovery portion is not necessary.